mmv1/products/bigqueryanalyticshub/DataExchangeSubscription.yaml

```diff
@@ -0,0 +1,250 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'DataExchangeSubscription'
+api_resource_type_kind: Subscription
+description: A Bigquery Analytics Hub Data Exchange subscription
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/bigquery/docs/analytics-hub-introduction'
+  api: 'https://cloud.google.com/bigquery/docs/reference/analytics-hub/rest/v1/projects.locations.subscriptions'
+docs:
+  note: |-
+    When importing the resource with `terraform import`, provide the destination project and location
+    in the format projects/{{destination_project}}/locations/{{destination_location}}/subscriptions/{{subscription_id}}
+  optional_properties: |
+    * `refreshWhenApply` - (Optional) If true, the subscription will be refreshed everytime Terraform apply is called
+    * `refreshWhenStale` - (Optional) If true, the subscription will be refreshed everytime the state is stale and Terraform apply is called
+base_url: 'projects/{{project}}/locations/{{location}}/subscriptions'
+self_link: 'projects/{{project}}/locations/{{location}}/subscriptions/{{subscription_id}}'
+create_url: 'projects/{{data_exchange_project}}/locations/{{data_exchange_location}}/dataExchanges/{{data_exchange_id}}:subscribe'
+import_format:
+  - 'projects/{{project}}/locations/{{location}}/subscriptions/{{subscription_id}}'
+custom_code:
+  decoder: 'templates/terraform/decoders/bigqueryanalyticshub_data_exchange_subscription.go.tmpl'
+  encoder: 'templates/terraform/encoders/bigqueryanalyticshub_data_exchange_subscription.go.tmpl'
+  post_read: 'templates/terraform/post_read/bigqueryanalyticshub_data_exchange_subscription.go.tmpl'
+  custom_update: 'templates/terraform/custom_update/bigqueryanalyticshub_data_exchange_subscription.go.tmpl'
+sweeper:
+  url_substitutions:
+    - region: "US"
+examples:
+  - name: 'bigquery_analyticshub_dataexchange_subscription_basic'
+    primary_resource_id: 'subscription'
+    primary_resource_name: 'fmt.Sprintf("tf_test_subscription_%s", context["random_suffix"])'
+    region_override: 'US'
+    vars:
+      data_exchange_id: 'my_test_dataexchange'
+      listing_dataset_id: 'listing_src_dataset'
+      listing_table_id: 'listing_src_table'
+      listing_id: 'my_test_listing'
+      subscription_id: 'my_subscription_id'
+      subscriber_contact_email: 'testuser@example.com'
+      # Variables for the Destination Dataset created by the Subscription
+      destination_dataset_id: 'subscribed_dest_dataset'
+      destination_dataset_friendly_name: 'Subscribed Destination Dataset'
+virtual_fields:
+  - name: 'refreshWhenApply'
+    type: Boolean
+    description: 'If true, the subscription will be refreshed everytime Terraform apply is called'
+    default_value: true
+  - name: 'refreshWhenStale'
```

Comment:
```suggestion
  - name: 'refresh_when_apply'
    type: Boolean
    description: 'If true, the subscription will be refreshed everytime Terraform apply is called'
    default_value: true
  - name: 'refresh_when_stale'
```

Reply:
better yet: please update our resource template to use snake case:
{{ $vf.Name }} -> {{ underscore $vf.Name }} on this line: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/resource.go.tmpl#L1172